File: StategyPattern/ModelDuck.cs
using StrategyPattern.Behaviors.FlyBehaviors;
using StrategyPattern.Behaviors.QuackBehavior;

namespace StrategyPattern
{
    internal class ModelDuck : Duck
    {
        public ModelDuck()
        {
            Flyebehavior = new FlyNoWay();
            QackBehavior = new MuteQuack();
        }

        public override void Display()
        {
            Console.WriteLine("model duck quack");
        }
    }
}

File: Strategy Pattern/Go/ducks/modelDuck.go
package ducks

import (
	"fmt"

	"github.com/rllko/Strategy-Pattern/behaviors/FlyBehaviors"
)

type modelDuck struct {
	flyBehavior FlyBehavior.IFlyBehavior
}

func NewModelDuck() *modelDuck {
	return &modelDuck{flyBehavior: FlyBehavior.FlyNowhere{}}
}

func (m *modelDuck) Fly() {
	m.flyBehavior.Fly()
}

func (m *modelDuck) Display() {
	fmt.Println("model duck quack")
}
